File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <strings.h>
#include <math.h>

int count_letters(char *string);
int count_sentences(char *string);
int count_words(char *string);

int main(void)
{
    char writing[100000];
    printf("Text: ");
    scanf("%[^\n]s", writing);

    float letters = count_letters(writing);
    float sentences = count_sentences(writing);
    float words = count_words(writing);

    float lettersperhund = (letters/words) * 100.00;
    float sentencesperhund = (sentences/words) * 100.00;

    int index = round((0.0588 * lettersperhund) - (0.296 * sentencesperhund) - 15.8);

    if(index < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }
    else if(index >= 1)
    {
        if(index >= 16)
        {
            printf("Grade 16+\n");
            return 0;
        }
        printf("Grade %d\n", index);
        return 0;
    }
    else if(index >= 16)
    {
        printf("Grade 16+\n");
        return 0;
    }
}

int count_letters(char *string)
{
    int numb = 0;
    char enders[8] = {'!', ',', '.', '?', ' ', '\'', ';','"'};
    for(int i = 0; i < strlen(string); i++)
    {
        for(int j = 0; j < sizeof(enders)/sizeof(enders[0]); j++)
        {
            if(string[i] == enders[j])
            {
                numb--;
                break;
            }
        }
        numb++;
    }
    return numb;
}
int count_sentences(char *string)
{
    int numb = 0;
    char enders[3] = {'!', '.', '?'};
    for(int i = 0; i <= strlen(string); i++)
    {
        for(int j = 0; j < sizeof(enders)/sizeof(enders[0]); j++)
        {
            if(string[i] == enders[j])
            {
                numb++;
                break;
            }
        }
    }
    return numb;
}
int count_words(char *string)
{
    int numb = 0;
    char enders[8] = {'!', ',', '.', '?', ' ', ';', '"'};
    for(int i = 0; i < strlen(string); i++)
    {
        for(int j = 0; j < sizeof(enders)/sizeof(enders[0]); j++)
        {
            if(string[i] == enders[j])
            {
                numb++;
                for(int k = 0; k < sizeof(enders)/sizeof(enders[0]); k++)
                {
                    if(string[i+1] == enders[k])
                    {
                        numb--;
                    }
                }
                break;
            }
        }
    }
    return numb + 1;
}

File: sentimental-readability/readability.py
# TODO

def main():
    passage = str(input("Enter your sentence: "))

    sentences = count_sentences(passage)
    letters = count_letters(passage)
    words = count_words(passage)

    s = float((sentences/words) * 100)
    l = float((letters/words) * 100)

    index = round((0.0588 * l) - (0.296 * s) - 15.8)


    if index <= 0:
        print("Before Grade 1.")
    elif index >= 16:
        print("Grade 16+")
    elif index < 16:
        print("Grade "+str(index))


def count_sentences(sentence):
    sentences = 0
    enders = {".", "?", "!"}
    for i in sentence:
        for j in enders:
            if i == j:
                sentences += 1
    return sentences

def count_letters(sentence):
    letters = 0
    enders = {".", "?", "!", " ", "/", ",", ";", ":","'"}
    for i in sentence:
        letters += 1
        for j in enders:
            if i == j:
                letters -= 1
    return letters

def count_words(sentence):
    words = 0
    length = 0
    enders = {".", "?", "!"," ", "/", ",", ";", ":"}
    for i in sentence:
        length += 1
        for j in enders:
            if i == j:
                words += 1
                for k in enders:
                    if length < len(sentence):
                        if sentence[length] == k:
                            words -= 1
    return words


if __name__ == "__main__":
    main()
